src/components/InputsTable.js

```diff
@@ -69,7 +70,12 @@ const InputsTable = props => {
   const parameterValueSelectWithWarnings = (rowIndex, selectedInputName) => {
     const warningMessage = Utils.cond(
       [missingRequiredInputs.includes(selectedInputName), () => 'This attribute is required'],
-      [inputsWithInvalidValues.includes(selectedInputName), () => 'Value is either empty or doesn\'t match expected input type'],
+      [inputsWithInvalidValues.includes(selectedInputName) && inputTableData[rowIndex].source.parameter_value === '', () => 'Value is empty'],
```

Comment:
Packing all of this into `inputsWithInvalidValues` seems to be having the side effects of:
1. Forcing you to override the "warning shape" and "warning color", in one specific situation, and
2. Reinterpreting the error messages later on to decide whether they mean we need to block submission or not

I wonder whether we could reframe this as "`inputsWithMessages`", a list of pairs/objects which look like `{messageType, message}`. Then:
1. It's obvious when we're rendering the message which icons/shapes we should use (and all errors/infos/validations/...) are all guaranteed to look the same
2. When we're checking whether any messages should block submission, we just need to check for "error" type messages. Or we could have a third `{isBlocker}` field on the objects.